manticore/ethereum.py

```diff
@@ -287,21 +287,100 @@ def __mul__(self, reps):
     SUINT = SByte(32)
     SValue = None
 
+    @staticmethod
+    def _tuple_bytes_to_long(tpl):
+        return reduce(lambda acc, b: (acc << 8) + ord(b), tpl, 0L)
+
+    class StaticBytes(object):
+        def __init__(self, value, size=32):
+            if not 0 < size <= 32:
+                raise ValueError('bytes<M>: binary type of M bytes, 0 < M <= 32.')
+            if isinstance(value, (int, long)):
+                if not 0 <= value < (2 ** (size * 8)):
+                    raise Value("value '%s' exceeds size %s" % (hex(value), size))
+                self.sub_word = ABI.serialize_uint(value, size)
+            elif isinstance(value, str) and value[:2] == '0x':
+                if not 3 <= len(value) <= 42:
+                    raise ValueError("hex '%s' does not fix in 32-byte bounds" % value)
+                # len(value) - 2 is the hex length
+                # (n + 1) // 2 gives the ceiling (accounts for odd length hex)
+                byte_len = (len(value) - 1) // 2
+                self.sub_word = ABI.serialize_uint(long(value, 16), byte_len)
+            elif isinstance(value, tuple):
+                if not 0 < len(value) <= 32:
+                    raise ValueError("value %s does not fix in 32-byte bounds" % value)
+                self.sub_word = value
+            else:
+                raise TypeError('Expected int, long, str, or tuple')
+
+        def size(self):
+            return len(self.sub_word)
+
+        def serialize(self):
+            return self.sub_word + tuple('\0' * (32 - len(self.sub_word)))
+
+    class Address(object):
```

Comment:
it seems this class is technically unnecessary b/c address is identical to uint160, as you noted in slack. in the name of minimizing the size of our codebase, i think i'd like to omit this class unless you feel strongly? i do understand it is used below in ExternalFunction, but it seems the main usage is in the type check, which i think could reasonably be left out